### PR TITLE
Added event tracking for chosen currency

### PR DESCRIPTION
**Pull request by Scollaco**
# 📲 What
The koala event tracking for Chosen Currency.

# 🤔 Why
Chosen Currency is part of `Settings v2`, but this event had been added on a PR containing events for `Settings v3`

# ✅ Acceptance criteria

- [x] I go to Settings > Account and change my currency by confirming the change on the Message dialog. The event `Selected Chosen Currency` should appear on Looker.
- [ ] The `Selected Chosen Currency`  should contain a property called `currency` with the name of the chosen currency. E.g: `€ Euro (EUR)`

please review my updated ACs (based on the conversation around testing Looker events) and either replace the original ones or discard these

Going to `Settings > Account` and changing the currency should
- [x] Track `Selected Chosen Currency` event in the console `🐨 [Koala Track]...`
- [x] `Selected Chosen Currency` event should contain a property called `currency` with the name of the chosen currency (e.g. `€ Euro (EUR)`).

### CODE REVIEW

**Comment by Scollaco** [comment]
[Add event for Chosen Currency](https://trello.com/c/xie662gI/1106-add-event-for-chosen-currency)

File: Library/ViewModels/SettingsAccountViewModel.swift [hunk 1]
```diff
@@ -88,6 +88,12 @@ SettingsAccountViewModelOutputs, SettingsAccountViewModelType {
 
     self.viewDidAppearProperty.signal
       .observeValues { _ in AppEnvironment.current.koala.trackAccountView() }
+
+    // Koala
+    updateCurrency.signal
+      .observeValues { currency in
```
Comment (dusi) [inline Library/ViewModels/SettingsAccountViewModel.swift]: Can we use 1 liner here? `.observeValues { AppEnvironment.current.koala.trackChangedCurrency($0) }` or is it better to be explicit?

File: Library/ViewModels/SettingsAccountViewModelTests.swift [hunk 2]
```diff
@@ -108,4 +108,16 @@ internal final class SettingsAccountViewModelTests: TestCase {
       XCTAssertEqual(["Viewed Account", "Viewed Account"], client.events)
     }
   }
+
+  func testTrackSelectedChosenCurrency() {
+
```
Comment (dusi) [inline Library/ViewModels/SettingsAccountViewModelTests.swift]: Do we need empty line here?

File: Library/Koala/KoalaTests.swift [hunk 3]
```diff
@@ -659,4 +659,13 @@ final class KoalaTests: TestCase {
...
     XCTAssertEqual(["Changed Password"], client.events)
   }
+
+  func testTrackChangedCurrency() {
+    let client = MockTrackingClient()
+    let koala = Koala(client: client)
+
+    koala.trackChangedCurrency(.CAD)
+
+    XCTAssertEqual(["Selected Chosen Currency"], client.events)
```
Comment (dusi) [inline Library/Koala/KoalaTests.swift]: Is there a reason why are we not testing properties here?

```suggestion
    XCTAssertEqual(["Selected Chosen Currency"], client.events)
    XCTAssertEqual(Currency.CAD.descriptionText, client.properties.last?["currency"] as? String)
```

Comment (Scollaco) [reply Library/Koala/KoalaTests.swift]: Good call. I tested on the SettingsAccountViewModel, but missed here.

File: Library/Koala/Koala.swift [hunk 4]
```diff
@@ -1320,6 +1320,10 @@ public final class Koala {
     self.track(event: "Resent Verification Email")
   }
 
+  public func trackChangedCurrency(_ currency: Currency) {
+    let prop = ["currency": currency.descriptionText]
+    self.track(event: "Selected Chosen Currency", properties: prop)
+  }
   /**
```
[the remark this answers is not in the document]
Comment (dusi) [reply Library/Koala/Koala.swift]: Was wondering, did you want to use this change or not? I've noticed the suggestion hasn't been applied.

File: Library/ViewModels/SettingsAccountViewModel.swift [hunk 5]
```diff
@@ -101,6 +101,12 @@ SettingsAccountViewModelOutputs, SettingsAccountViewModelType {
 
     self.viewDidAppearProperty.signal
       .observeValues { _ in AppEnvironment.current.koala.trackAccountView() }
+
+    // Koala
+    updateCurrency.signal
+      .observeValues {
+        AppEnvironment.current.koala.trackChangedCurrency($0)
```
Comment (dusi) [inline Library/ViewModels/SettingsAccountViewModel.swift]: This is fine...I think my previous comment was that we could have had this on 1 line like so..

```
updateCurrency.signal
  .observeValues { AppEnvironment.current.koala.trackChangedCurrency($0) }
```

^ up to you

Comment (Scollaco) [reply Library/ViewModels/SettingsAccountViewModel.swift]: Oh ok. I particularly doesn't have preference, but changed it to have it in 1 line. Less one line to read 😄

**Pull request merged**